File: src/com/designpatterns/creational/abstractfactory/RegularShapeFactory.java
package com.designpatterns.creational.abstractfactory;

public class RegularShapeFactory implements IShapeFactory {

    public static final String TAG = "RegularShapeFactory";

    @Override
    public IShape getShape(String shapeType) {
        if (shapeType == null) {
            return null;
        }
        if (shapeType.equalsIgnoreCase(Rectangle.TAG)) {
            return new Rectangle(4.0f, 5.0f);
        } else if (shapeType.equalsIgnoreCase(Square.TAG)) {
            return new Square(5.0f);
        } else {
            throw new IllegalArgumentException("Regular shape type not found");
        }
    }
}

File: src/com/designpatterns/creational/abstractfactory/kotlin/RegularShapeFactory.kt
package com.designpatterns.creational.abstractfactory.kotlin

import java.lang.IllegalArgumentException

class RegularShapeFactory : IShapeFactory {

    override fun getShape(shapeType: String): IShape {

        return if (shapeType.equals(Rectangle.TAG, ignoreCase = true)) {
            Rectangle(4.0f, 5.0f)
        } else if (shapeType.equals(Square.TAG, ignoreCase = true)) {
            Square(5.0f)
        } else {
            throw IllegalArgumentException("Regular shape type not found")
        }
    }

    companion object {
        const val TAG = "RegularShapeFactory"
    }
}